### ISSUE TITLE
Support retrieving TOTP via CLI

### ISSUE THREAD

**Issue by njgraf512** [lastpass/lastpass-cli#592]
The CLI doesn't seem to support retrieving stored TOTP's. Is this a feature that's being added?

**Comment by MacLeodMike** [comment]
The ability to set TOTP via CLI would also be extraordinarily useful.

**Comment by qwinkler** [comment]
Any updates here?

**Comment by schmitmd** [comment]
The project is dead @qwinkler . See #602 . You might be able to convince someone who's helping to maintain the fork mentioned in that thread, but I wouldn't bother with this repo.   R.I.P.

**Comment by qwinkler** [comment]
@schmitmd Sounds funny, haha
I decided to use another util, but it is so weird, that the LastPass just bailed on its users.

Anyway, thanks for the answer, I appreciate it. And have a nice day 😄

**Comment by schmitmd** [comment]
Yeah, I'm really sad about it.  Had to move to another util after being a LastPass user for like, 10+ years because this one tool was absolutely essential to me.

**Comment by robwithhair** [comment]
We have a business account.  We need to be able to access creds on servers with no UI.  Would be good to get this feature so that we can continue to use LastPass.

**Comment by snaeqe** [comment]
I am a happy user of lastpass-cli and creating TOTP tokens out of it.
Yes, you have to use a workaround, but IMHO that works beautifully.
I am using https://github.com/bfreis/otp to generate the tokens, and for secret storage, I use the password field from a separate entry in LastPass.

So in the end, the secret sauce is nothing more than
`otp -s $(lpass show --password <secret-identifier>`
and there's your token.

**Comment by haegarAI** [comment]
> I am a happy user of lastpass-cli and creating TOTP tokens out of it. Yes, you have to use a workaround, but IMHO that works beautifully. I am using https://github.com/bfreis/otp to generate the tokens, and for secret storage, I use the password field from a separate entry in LastPass.
> 
> So in the end, the secret sauce is nothing more than `otp -s $(lpass show --password <secret-identifier>` and there's your token.

_As we have some special characters in the secret, this way - using base32 conversion- will hopefully work:_ ,,,sorry this was nonsense... I mixed up secret and password... as I was not the one who created this entry in LastPass and I never did a 2FA setup in Lastpass until today ;) 
As all my tries failed to create the same 2FA TOTP like Lastpass it finally came to my mind, that I would need to use the account to login to Lastpass, to get the "secret" ... which is in fact the needed base32 key ;)... I feel so dumb :(...

But I also learned sth.: the person who created the 2FA still has this secret key maybe, because this was never changed! Ofc he would need the password as well, which we changed several times since he has left the company... but anyway, this is also sth which should be regenerated, when someone leaves the company!

This easy to copy/bash snippet2go was on my way to enlightening:
~~~
l:~$ cat totp.sh 
#!/bin/bash
count="$(printf '%.16x' $(($(date +%s)/30)))"
hexkey="$(echo -n "$1" | base32 -d | xxd -p)"
hash="$(echo -n "$count" | xxd -r -p | openssl mac -digest sha1 -macopt hexkey:"$hexkey" HMAC)"
offset="$((16#${hash:39}))"
extracted="${hash:$((offset * 2)):8}"
echo "$(((16#$extracted & 16#7fffffff) % 1000000))"
~~~
thanks to https://thenybble.de/posts/calculating-totp/

and yes for using this nice bash script you will need the shared secret of the 2FA , which is usualy shown only once when you pair your auth app with the 2FA of an account with password on some website. 
In Lastpass you can view it later also here:

<img width="731" height="543" alt="Image" src="https://github.com/user-attachments/assets/fb0b5d64-ef16-47a1-a1cc-7380c2def709" /> clicking "view" next to 'Barcode' (which is in fact a QR code) , will show the QR code which you can scan with QR code reader on your phone, where you can then read the base32 code (=secret) .. Tadaaa!! ;) 

for all lost souls out there, like me ;) ...
